panic: runtime error: invalid memory address or nil pointer dereference

# What did you do? (required. The issue will be **closed** when not provided.)

Updated below:
Shows the updated version.

- go version go1.15.6 linux / amd64
- go-cpe-dictionary v0.2.5 3183fd0
- go-cve-dictionary v0.5.6 c01cf7c
- goval-dictionary v0.3.0 af24f65
- go-exploitdb v0.1.2
- vuls v0.15.2 build-20210114_153713_dfffe5b

Other version information is as follows:
OS:
CentOS Linux release 7.3.1611 (Core)

SQLite3:
3.7.17 2013-05-20 00:56:22 118a3b35693b134d56ebd780123b7fd6f1497668

gost:
gost 3060e46

# What did you expect to happen?

As before the update, I expected the email to reach the address registered in `config.toml` as before with` vuls report -to-email` option.

# What happened instead?

An error occurred and the email was not sent.

* Current Output

```
Vuls Scanning [Project-A] starts ...
[Jan 15 12:24:17] INFO [localhost] Start scanning
[Jan 15 12:24:17] INFO [localhost] config: /home/vuls/config/Project-A/config.toml
[Jan 15 12:24:17] INFO [localhost] Validating config ...
[Jan 15 12:24:17] INFO [localhost] Detecting Server / Container OS ...
[Jan 15 12:24:17] INFO [localhost] Detecting OS of servers ...
[Jan 15 12:24:17] INFO [localhost] (1/1) Detected: Project-A: pseudo
[Jan 15 12:24:17] INFO [localhost] Detecting OS of containers ...
[Jan 15 12:24:17] INFO [localhost] Checking Scan Modes ...
[Jan 15 12:24:17] INFO [localhost] Detecting Platforms ...
[Jan 15 12:24:17] INFO [localhost] (1/1) Project-A is running on other
[Jan 15 12:24:17] INFO [localhost] Detecting IPS identifiers ...
[Jan 15 12:24:17] INFO [localhost] (1/1) Project-A has 0 IPS integration
[Jan 15 12:24:17] INFO [Project-A] Scanning listen port ...


Scan Summary
================
Project-A pseudo 0 installed, 0 updatable

To view the detail, vuls tui is useful.
To send a report, run vuls report -h.

Vuls Reporting [Project-A] starts ...
[Jan 15 12:24:17] INFO [localhost] Validating config ...
[Jan 15 12:24:17] INFO [localhost] Loaded: / home / vuls / results / Project-A / 2021-01-15T12: 24: 17 + 09: 00
[Jan 15 12:24:17] INFO [localhost] Validating db config ...
INFO [0000] -cvedb-type: sqlite3, -cvedb-url:, -cvedb-path: /home/vuls/cve.sqlite3
INFO [0000] -ovaldb-type: sqlite3, -ovaldb-url:, -ovaldb-path: /home/vuls/oval.sqlite3
INFO [0000] -gostdb-type: sqlite3, -gostdb-url:, -gostdb-path: /home/vuls/gost.sqlite3
INFO [0000] -exploitdb-type: sqlite3, -exploitdb-url:, -exploitdb-path: /home/vuls/go-exploitdb.sqlite3
INFO [0000] -msfdb-type: sqlite3, -msfdb-url:, -msfdb-path: /home/vuls/go-msfdb.sqlite3
[Jan 15 12:24:17] WARN [localhost] --ovaldb-path = / home / vuls / oval.sqlite3 file not found
[Jan 15 12:24:17] WARN [localhost] --gostdb-path = / home / vuls / gost.sqlite3 file not found. Vuls can detect `patch-not-released-CVE-ID` using gost if the scan target server is Debian, RHEL or CentOS, For details, see `https://github.com/knqyf263/gost#fetch-redhat`
[Jan 15 12:24:17] WARN [localhost] --msfdb-path = / home / vuls / go-msfdb.sqlite3 file not found. Fetch go-msfdb before reporting if you want to display metasploit modules of detected CVE- IDs. For details, see `https://github.com/takuzoo3868/go-msfdb`
[Jan 15 12:24:17] INFO [localhost] pseudo type. Skip OVAL and gost detection
[Jan 15 12:24:20] INFO [localhost] Project-A: 239 CVEs are detected with CPE
[Jan 15 12:24:20] INFO [localhost] Fill CVE detailed with gost
[Jan 15 12:24:20] INFO [localhost] Fill CVE detailed with CVE-DB
[Jan 15 12:24:21] INFO [localhost] Fill exploit with Exploit-DB
[Jan 15 12:24:21] INFO [localhost] Project-A: 11 exploits are detected
[Jan 15 12:24:21] INFO [localhost] Fill metasploit module with Metasploit-DB
[Jan 15 12:24:21] INFO [localhost] Project-A: 0 modules are detected
[Jan 15 12:24:21] INFO [localhost] Fill CWE with NVD
[Jan 15 12:24:21] INFO [localhost] Previous json found: /home/vuls/results/Project-A/2021-01-15T11:29:34+09: 00/Project-A.json
[Jan 15 12:24:21] INFO [localhost] Project-A: There are 239 vulnerabilities, but no difference between current result and previous one.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code = 0x1 addr = 0x20 pc = 0xd3c90f]

goroutine 1 [running]:
github.com/emersion/go-smtp.(*Client).Auth (0xc0000db760, 0x0, 0x0, 0x0, 0x0)
        /home/vuls/go/pkg/mod/github.com/emersion/go-smtp@v0.14.0/client.go:291 + 0x6f
github.com/future-architect/vuls/report. (* EmailSender) .sendMail (0xc000c281b0, 0xc00025dd40, 0xc, 0xc00064d680, 0x206, 0x0, 0x0)
        /home/vuls/go/src/github.com/future-architect/vuls/report/email.go:128 +0x35a
github.com/future-architect/vuls/report.(*emailSender). Send (0xc000c281b0, 0xc00073bf40, 0x3f, 0xc0002a7ce0, 0xd6, 0xc0001acbf8, 0x5)
        /home/vuls/go/src/github.com/future-architect/vuls/report/email.go:191 + 0x8bb
github.com/future-architect/vuls/report.EMailWriter.Write (0xc000178000, 0x1, 0x1, 0x0, 0x0)
        /home/vuls/go/src/github.com/future-architect/vuls/report/email.go:53 + 0xec7
github.com/future-architect/vuls/subcmds. (* ReportCmd) .Execute (0xc00040d580, 0x17b0ce0, 0xc0000b2010, 0xc00007e900, 0x0, 0x0, 0x0, 0x0)
        /home/vuls/go/src/github.com/future-architect/vuls/subcmds/report.go:348 +0x10bb
github.com/google/subcommands. (* Commander) .Execute
```


# Steps to reproduce the behaviour

```
PROJECT = "Project-A"
CONFIG = "/ home / vuls / config / $ PROJECT / config.toml"
RESULT = "/ home / vuls / results / $ PROJECT"
REPORT = "/ home / vuls / reports / $ PROJECT"
LOGDIR = "/ var / log / vuls / $ PROJECT"
FORMAT = "-format-full-text"
DIFF = "-diff"
TODAY = "` date +% Y% m% d` "

#Detect scan
echo "Vuls Scanning [" $ PROJECT "] starts ..."
vuls scan -config = $ CONFIG -results-dir = $ RESULT -log-dir = $ LOGDIR

#Report report
echo "Vuls Reporting [" $ PROJECT "] starts ..."
vuls report -format-json $ FORMAT -lang ja $ DIFF -config = $ CONFIG -results-dir = $ RESULT -log-dir = $ LOGDIR -to-email> "$ REPORT / Vuls-Report-["$ PROJECT "" ] "$ FORMAT"-"$ TODAY $ DIFF" .txt "
echo echo
```

# Configuration (**MUST** fill this out):

* Go version (`go version`):
go version go1.15.6 linux / amd64

* Go environment (`go env`):

```
GO111MODULE = ""
GOARCH = "amd64"
GOBIN = ""
GOCACHE = "/ home / vuls / .cache / go-build"
GOENV = "/ home / vuls / .config / go / env"
GOEXE = ""
GOFLAGS = ""
GOHOSTARCH = "amd64"
GOHOSTOS = "linux"
GOINSECURE = ""
GOMODCACHE = "/ home / vuls / go / pkg / mod"
GONOPROXY = ""
GONOSUM DB = ""
GOOS = "linux"
GOPATH = "/ home / vuls / go"
GO PRIVATE = ""
GOPROXY = "https://proxy.golang.org, direct"
GOROOT = "/ usr / local / go"
GOSUMDB = "sum.golang.org"
GOTMPDIR = ""
GOTOOLDIR = "/ usr / local / go / pkg / tool / linux_amd64"
GCCGO = "gccgo"
AR = "ar"
CC = "gcc"
CXX = "g ++"
CGO_ENABLED = "1"
GOMOD = ""
CGO_CFLAGS = "-g -O2"
CGO_CPPFLAGS = ""
CGO_CXXFLAGS = "-g -O2"
CGO_FFLAGS = "-g -O2"
CGO_LDFLAGS = "-g -O2"
PKG_CONFIG = "pkg-config"
GOGCCFLAGS = "-fPIC -m64 -pthread -fmessage-length = 0 -fdebug-prefix-map = / tmp / go-build732032478 = / tmp / go-build -gno-record-gcc-switches"
```

* Vuls environment:

vuls v0.15.2 build-20210114_153713_dfffe5b

* config.toml:

```
[servers]

[slack]
hookURL = "https://outlook.office.com/webhook/xxxx
channel = "#VulsReport"
authUser = "VulsReporter"

[servers.Project-A]
type = "pseudo"
cpeNames = [
"cpe: / o: netbsd: netbsd:-",
"cpe: / a: zlib: zlib: 1.1.4",
"cpe: / a: info-zip: unzip: 5.52",
"cpe: / a: info-zip: wiz: 2.31",
"cpe: / a: microsoft: sql_server: 2017:-: ~~~~ x64 ~",
"cpe: / o: microsoft: windows_10: 1607 :: ~~~~ x64 ~",
"cpe: / a: microsoft: .net_framework: 3.5.1",
"cpe: / a: microsoft: .net_framework: 4.7.2",
]

[email]
smtpAddr = "localhost"
smtpPort = "25"
# user = "xxxx"
# password = "xxxxxxxx"
from = "vuls@xxxx.com"
to = [
"aaaa@xxxx.com",
"bbbb@xxxx.com",
"cccc@xxxx.com",
"dddd@xxxx.com",
"eeee@xxxx.com",
]
cc = [
"ffff@xxxx.com",
"gggg@xxxx.com",
"hhhh@xxxx.com",
"iiii@xxxx.com",
"jjjj@xxxx.com",
"kkkk@xxxx.com",
# "llll@xxxx.com",
# "mmmm@xxxx.com",
]
subjectPrefix = "[VulsReport]"
```

--- a/report/email.go
+++ b/report/email.go
@@ -123,11 +123,11 @@ func (e *emailSender) sendMail(smtpServerAddr, message string) (err error) {
 	if ok, param := c.Extension("AUTH"); ok {
 		authList := strings.Split(param, " ")
 		auth = e.newSaslClient(authList)
+		if err = c.Auth(auth); err != nil {
+			return xerrors.Errorf("Failed to authenticate: %w", err)
+		}
 	}
 
-	if err = c.Auth(auth); err != nil {
-		return xerrors.Errorf("Failed to authenticate: %w", err)
-	}
 	if err = c.Mail(emailConf.From, nil); err != nil {
 		return xerrors.Errorf("Failed to send Mail command: %w", err)
 	}

